README.rst

```diff
@@ -82,70 +84,78 @@ PyPI wheels (pip)
+where ``${PL_BACKEND}`` can be ``lightning_qubit`` (default), ``lightning_gpu``,  ``lightning_kokkos``, or ``lightning_tensor``.
 The `pybind11 <https://pybind11.readthedocs.io/en/stable/>`_ library is required to bind the C++ functionality to Python.
 
 A C++ compiler such as ``g++``, ``clang++``, or ``MSVC`` is required.
 On Debian-based systems, this can be installed via ``apt``:
 
-.. code-block:: console
+.. code-block:: bash
 
-    $ sudo apt -y update &&
-    $ sudo apt install g++ libomp-dev
+    sudo apt -y update && sudo apt install -y g++ libomp-dev
 
 where ``libomp-dev`` is included to also install OpenMP.
 On MacOS, we recommend using the latest version of ``clang++`` and ``libomp``:
 
-.. code-block:: console
+.. code-block:: bash
 
-    $ brew install llvm libomp
+    brew install llvm libomp
 
-The Lightning-GPU backend has several dependencies (e.g. ``CUDA``, ``custatevec-cu12``, etc.), and hence we recommend referring to Lightning-GPU installation section.
-Similarly, for Lightning-Kokkos it is recommended to configure and install Kokkos independently as prescribed in the Lightning-Kokkos installation section.
+The Lightning-GPU backend has several dependencies (e.g. ``CUDA``, ``custatevec-cu12``, etc.), and hence we recommend referring to `Lightning-GPU installation <https://docs.pennylane.ai/projects/lightning/en/stable/lightning_gpu/installation.html>`_ section.
+Similarly, for Lightning-Kokkos it is recommended to configure and install Kokkos independently as prescribed in the `Lightning-Kokkos installation <https://docs.pennylane.ai/projects/lightning/en/stable/lightning_kokkos/installation.html>`_ section.
 
 Development installation
 ========================
 
 For development and testing, you can install by cloning the repository:
 
-.. code-block:: console
+.. code-block:: bash
 
-    $ git clone https://github.com/PennyLaneAI/pennylane-lightning.git
-    $ cd pennylane-lightning
-    $ pip install -r requirements.txt
-    $ PL_BACKEND=${PL_BACKEND} python scripts/configure_pyproject_toml.py
-    $ pip install -e . -vv
+    git clone https://github.com/PennyLaneAI/pennylane-lightning.git
+    cd pennylane-lightning
+    pip install -r requirements.txt
+    PL_BACKEND=${PL_BACKEND} python scripts/configure_pyproject_toml.py
+    pip install -e . -vv
 
 Note that subsequent calls to ``pip install -e .`` will use cached binaries stored in the
 ``build`` folder, and the ``pyproject.toml`` file defined by the configuration script. Run ``make clean`` if you would like to recompile from scratch.
 
 You can also pass ``cmake`` options with ``CMAKE_ARGS`` as follows:
 
-.. code-block:: console
+.. code-block:: bash
+
+    CMAKE_ARGS="-DENABLE_OPENMP=OFF -DENABLE_BLAS=OFF" pip install -e . -vv
+    
+Supported options are
+
+- ``-DENABLE_WARNINGS:BOOL=ON``
+- ``-DENABLE_NATIVE:BOOL=ON`` (for ``-march=native``)
+- ``-DENABLE_BLAS:BOOL=ON``
+- ``-DENABLE_OPENMP:BOOL=ON``
+- ``-DENABLE_CLANG_TIDY:BOOL=ON``
```

Comment:
Would you like to add details for each of these options? If not, I may suggest review them in one line

```
Supported options are ``ENABLE_WARNINGS``, ``ENABLE_NATIVE`` (for ``-march=native``) ``ENABLE_BLAS``, ``ENABLE_OPENMP``,  and ``ENABLE_CLANG_TIDY``.
```
Note that that not all of these options are enabled by default.

Reply:
Come to me another idea :bulb::
List all the Cmake options and put them in a file like the following:
![image](https://github.com/user-attachments/assets/067301ea-708c-41ee-b94d-b67a62b0b916)
...

And adding the link to the file in the description as follows:
```
The supported options for CMakes are listed in the file [CMakeOptions.rst](doc/CMakeOptions.rst).
```

However, I do not know where to put this file. Would be right to add it inside the `doc` directory? :thinking:

Reply:
I love it! However, this list already exists in CMakeLists.txt and this is not the list of all options! Only the ones that we have explored or added so far. I'm not sure if this is worth adding to public docs as it requires more efforts on keeping this list up-to-date. We also don't need to add `Kokkos_` specific options to readme; only provide links to their docs/guidelines and it should be sufficient. Let's not make our README so long :smile:

Reply:
Therefore I will do these steps
* remove the `CMakeOptions.rst` from `doc` dir 
* Replace by 
```
Supported options are ``ENABLE_WARNINGS``, ``ENABLE_NATIVE`` (for ``-march=native``) ``ENABLE_BLAS``, ``ENABLE_OPENMP``,  and ``ENABLE_CLANG_TIDY``.
```

However, I think there is not enough information for the Cmake options. I suggest finding and alternative for beginner developers. I agree to not create a so long README. To keep updated on this list we can do the following command
``` bash
grep -r --exclude-dir=_deps 'option(' . | awk -F ':' '{print $2}' | sed -e 's/option(// ; s/#//' | tr -d ')' | awk '{$1=$1;print}' | sort | awk -F '"' '{printf "-D%-40s %s\n", $1"="$3, $2 }' | sed -e 's/ = /=/'
```